src/CodeRepositoryProviders/BitbucketCodeRepositoryProvider.php

```diff
@@ -4,20 +4,67 @@
 
 namespace App\CodeRepositoryProviders;
 
-use CodeRepository;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
 
 class BitbucketCodeRepositoryProvider implements Provider
 {
-    const NAME = 'bitbucket';
-
-    public function fetch(FetchCriteria $name): iterable
+    public function __construct(private HttpClientInterface $httpClient)
     {
-        if ($name->providerName !== self::NAME) {
-            return [];
+    }
+
+    /**
+     * @return CodeRepository[]
+     */
+
+        public function fetch(FetchCriteria $criteria): iterable
+//        {
+//            if ($name->providerName !== self::NAME) {
+//                return [];
+//            }
+        {
+            $response = $this->httpClient->request('GET', "https://api.bitbucket.org/2.0/repositories/$criteria->organizationName?page=1&per_page=100");
```

Comment:
I would fix some formatting:

```suggestion
      /**
       * @return CodeRepository[]
       */
        public function fetch(FetchCriteria $criteria): iterable
        {
            $response = $this->httpClient->request('GET', "https://api.bitbucket.org/2.0/repositories/$criteria->organizationName?page=1&per_page=100");
```